ipfx/nwb_reader.py

```diff
@@ -415,24 +410,43 @@ def get_stim_code(self, sweep_name):
 
 def get_nwb_version(nwb_file):
     """
-    Return a dict with `major` and `full` NWB version as read from the NWB
-    file.
+    Return a dict with `major` and `full` NWB version as read from the NWB file.
     """
 
     with h5py.File(nwb_file, 'r') as f:
-        # In v1 this is a dataset
-        nwb_version = f["/"].get("nwb_version")
-        if nwb_version is not None and re.match("^1", nwb_version):
-            return {"major": 1, "full": nwb_version}
-
-        # but in V2 this is an attribute
-        nwb_version = f["/"].attrs.get("nwb_version")
-        if nwb_version is not None and re.match("^2", nwb_version):
-            return {"major": 2, "full": nwb_version}
+        if "nwb_version" in f:         # In v1 this is a dataset
+            nwb_version = f["nwb_version"].value
+            if isinstance(nwb_version, np.ndarray):
+                nwb_version = nwb_version[0]
+            if nwb_version is not None and re.match("NWB-1", nwb_version):
+                return {"major": 1, "full": nwb_version}
+
+        elif "nwb_version" in f.attrs:   # but in V2 this is an attribute
+            nwb_version = f.attrs["nwb_version"].value
+            if nwb_version is not None and re.match("^2", nwb_version):
+                return {"major": 2, "full": nwb_version}
 
     return {"major": None, "full": None}
 
 
+def get_nwb_flavor(nwb_file):
+
+    with h5py.File(nwb_file, 'r') as f:
+        sweep_names = [e for e in f["acquisition/timeseries"].keys()]
+        sweep_naming_convention = sweep_names[0].split('_')[0]
+
+        if sweep_naming_convention == "Sweep":
+            nwb_flavor = "Pipeline"
+
+        elif sweep_naming_convention == "data":
+            nwb_flavor = "Mies"
+
+        else:
+            raise ValueError("Unknown sweep naming convention: %s" %sweep_naming_convention)
```

Comment:
flake8 complains here with `nwb_reader.py:445:69: E225 missing whitespace around operator`. Not sure if you follow these rules here or not.